### ISSUE TITLE
Add rnpm like plugin support to Haul

### ISSUE THREAD

**Issue by satya164** [callstack-io/haul#235]
It'd be great to be able to add additional commands to haul like in rnpm. We could have a separate plugin which provides a single command to automate publishing to codepush etc.

**Comment by grabbou** [comment]
Yeah, it doesn't take a lot of lines, can backport from react-native.

**Comment by grabbou** [comment]
Let's wait for someone to raise this use-case, for now, it's icebox.

**Comment by grabbou** [comment]
CC: @zamotany since you mentioned some plugins proposal.
